### ISSUE TITLE
Improve the reliability of the update process

### ISSUE THREAD

**Issue by UriHerrera** [Nitrux/nuts#6]
The original idea of the Nitrux Update Tool System is straightforward, as described in the README.

1. Creates a backup of the root directory using SquashFS and the XFS partition and stores them locally.
2. Download the latest ISO image using the BitTorrent protocol and update the system using rsync.
3. When restoring a backup, use the locally generated SquashFS file (instead of downloading an ISO).

However, for some reason I've yet to understand while this premise worked during the creation and initial testing of the utility, it seems to no longer be an effective way of updating the root directory. While the utility presents no problems, the resulting root directory post-update remains non-functional. This indicates that the utility is performing its tasks, but the logic behind how rsync is used may be wrong.

As of the latest release, version 3.0.0, and after upgrading from 2.9.1, the result is that all the partitions except the root partition are not being mounted, or the Wayland session has become unusable; see [https://github.com/Nitrux/nitrux-bug-tracker/issues/129](https://github.com/Nitrux/nitrux-bug-tracker/issues/129).  That is, of course, undesired behavior.

So, with that said, there needs to be a more reliable way of replacing the root directory that still fits within the vision and scope of this utility.

**Comment by UriHerrera** [comment]
One of the issues noted above is fixed now, specifically that partitions except the root partition were not being mounted. However, the other issue remains: the Wayland session does not work after upgrading. But there's a twist to this.

While troubleshooting that problem, I've found that after the utility completes the upgrade process and reboots in the status described in the issue, restoring the backup SquashFS that the utility creates during the update process makes the Wayland session work again. I find this behavior strange because, in either case, a SquashFS is used, and `rsync` is also used along with the same exclusion list, yet that occurs.

The SquashFS in the ISO file does work, but I can't narrow down the file(s) causing this problem; as far as I've dug, it's some file in /usr, maybe a library. I've tried multiple ways of excluding files or directories from `rsync` to no avail. 

An approach I've decided to try is to replicate the restore process in the update process, i.e., use a backup SquashFS from a newer installation.

**Comment by UriHerrera** [comment]
Fixed in 339527a

It turns out that the above was the better solution. Instead of _excluding_ things when using rsync, create a SquashFS that _includes_ what we want to update. To create these files, I also created a tool called [mkota](https://github.com/Nitrux/storage/blob/master/Other/mkota). When using `restore` though, the exclusion list is still used since that was already working.

As usual, the more straightforward solution is the best 🤷🏾 

Also, no more aria2c, it's all axel. So, downloads would be faster.

TL;DR, this utility no longer downloads an ISO, mounts it, then mounts the SquashFS, then syncs; instead, it downloads a SquashFS with only the updated stuff included, mounts it, and syncs.

**Comment by UriHerrera** [comment]
I've reopened this issue because I've tried with further SquashFS archives without consistent results. One moment, the archive works, then sometime later, applying it again results in a non-functional installation.

I must consider rewriting this utility in a programming language like Python because shell script works for most tasks but falls short in others. Still, updating the root is consistently proving difficult to do only using shell script.

**Comment by UriHerrera** [comment]
TL;DR Fixed in https://github.com/Nitrux/nuts/commit/09e991fd3c7188a953ebba4787dc3dc955df6d84

Much testing, very refactored. The utility works to upgrade, restore (rollback), self-update, and rescue was working already; stuff is not broken.

Success, closing this issue.
